source/renderer/app/components/wallet/WalletSendForm.js

```diff
@@ -1016,7 +1017,7 @@ export default class WalletSendForm extends Component<Props, State> {
                 label={intl.formatMessage(messages.addAssetButtonLabel)}
                 disabled={!this.hasAvailableAssets}
                 onClick={() => {
-                  this.setState({ isTokenPickerOpen: true });
+                  this.walletTokenPickerRef?.current?.open?.();
```

Comment:
Why is it needed to use `ref` instead of `state`? This is creating a dependency between WalletSendForm and `WalletTokenPickerDialogContainer` as it requires the private method `open` to exist.

Reply:
The `open` method in this case is part of a public API of the `walletTokenPickerRef` component so I wouldn't call it private. It is just exposed through the ref API. The reason I resigned from local state is that I wanted to base the mechanic of opening/closing the dialog on the global `uiDialogs` state. Alternatively I could implement the open method on the closest parent container component and pass it down to here which is also fine for me. The reason I chose this approach is that it encapsulates all dialog related logic in the dedicated dialog component. Also I had the `WalletTokenPicker` component already imported there (it is needed for the uiDialogs' open function). However I understand it is not common to use ref apis. If this is problematic I am fine to refactor it to passing the `open` function from above.